multi/imxrt-multi/uart.c

```diff
@@ -194,28 +194,24 @@ static void set_cflag(void *_uart, tcflag_t* cflag)
 }
 
 
-static uint32_t calculate_baudrate(speed_t baud)
+static uint32_t calculate_baudrate(uint32_t baud)
 {
-	uint32_t osr, sbr, bestSbr = 0, bestOsr = 0, bestDiff, t, tDiff, baud_rate;
+	uint32_t osr, sbr, bestSbr = 0, bestOsr = 0, bestDiff, t, tDiff;
 
-	bestDiff = baud_rate = (uint32_t)libtty_baudrate_to_int(baud);
-
-	if ((int)baud_rate <= 0) {
-		return 0;
-	}
+	bestDiff = baud;
 
 	for (osr = 4; osr <= 32; ++osr) {
 		/* find sbr value in range between 1 and 8191 */
-		sbr = (UART_CLK / (baud_rate * osr)) & 0x1fff;
+		sbr = (UART_CLK / (baud * osr)) & 0x1fff;
```

Comment:
If `baud=0` division by zero. The removed `if` check shall be restored/updated for that case.

Reply:
I have updated this check.